### PR TITLE
Basic auth login istio routing rule

### PR DESCRIPTION
**Pull request by gabrielwen**
Relates to #2766

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/3083)

### CODE REVIEW

**Comment by kunmingg** [comment]
When will we remove ambassador annotations?

**Comment by gabrielwen** [comment]
after we remove ambassador as component.

File: kubeflow/common/basic-auth.libsonnet [hunk 1]
```diff
@@ -41,6 +41,48 @@
     },
     authService:: authService,
 
+    local authIstioVirtualService = {
```
[the remark this answers is not in the document]
Comment (gabrielwen) [reply kubeflow/common/basic-auth.libsonnet]: updated.

**Pull request merged**